pkg/resource/generator.go

```diff
@@ -285,5 +287,22 @@ func (g *Generator) Remove(cr *imageregistryv1.Config, modified *bool) error {
 		*modified = true
 	}
 
+	driver, err := storage.NewDriver(cr.ObjectMeta.Name, cr.ObjectMeta.Namespace, &cr.Status.Storage)
+	if err != nil {
+		return err
+	}
+
+	var derr error
+	err = wait.PollImmediate(1*time.Second, 5*time.Minute, func() (stop bool, err error) {
+
+		if derr = driver.RemoveStorage(cr, modified); derr != nil {
+			return false, nil
```

Comment:
please do a follow up to make this more fine-grained.  We do not want to wait 5 minutes to remove storage in the face of fatal errors.

Reply:
fine-grained as in check the type of error that we get? or fine-grained as in a lower timeout?

Reply:
check the type of error.  So we don't spend 5 minutes retrying this operation if we (should) know it's never going to succeed.

Reply:
updated

Reply:
If an aws error returns, it's not retriable, else it is retriable (updated in the RemoveStorage function)